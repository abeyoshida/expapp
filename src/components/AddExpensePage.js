import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  
  onSubmit = (expense) => {
    this.props.addExpense(expense);  // this will dispatch the addExpense call with the expense argument
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    )
  }
};

const mapDispatchToProps = (dispatch) => ({
   // the mapDispatchToProps passes the dispatch call to addExpense
   // addExpense will now be available to the component via this.props.addExpense
  addExpense: (expense) => dispatch(addExpense(expense)) 
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
