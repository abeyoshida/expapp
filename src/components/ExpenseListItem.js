import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  //const createdAtSafe = (moment.isMoment(createdAt)) ? createdAt.format("dddd, MMMM Do YYYY, h:mm:ss a") : createdAt;
  const createdAtSafe = '';
  return (
      <div>
        <Link to={`/edit/${id}`}>
          <h3>{description}</h3>
        </Link>
        {/*<p>{amount} - {createdAt}</p>*/}
        <p>{amount} - {createdAtSafe}</p>
      </div>
    );
}
export default ExpenseListItem;
