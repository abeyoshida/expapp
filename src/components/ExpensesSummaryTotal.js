import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

class ExpensesSummaryTotal extends React.Component {
    constructor(props) {
        super(props);
    };
    
    render() {
        return (
            <div>
                {
                    (this.props.expensesCount === 0) ? <p>There are no expenses</p> : <p>The total amount of your expenses: {this.props.expensesTotal}</p>
                }
                
            </div>
        )
    }
};

const mapStateToProps = (state) => {
    const expensesTotal = selectExpensesTotal(state.expenses);
    return {
        expenses: selectExpenses(state.expenses, state.filters),
        expensesCount: state.expenses.length,
        expensesTotal: numeral(expensesTotal/100).format('$0,0.00')
    }
};

export default connect(mapStateToProps)(ExpensesSummaryTotal);