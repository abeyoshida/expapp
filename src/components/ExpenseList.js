import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

// this first named export is for testing an version that is not connected to the store
export const ExpenseList = (props) => {
  //console.log('ExpenseList.js ExpenseList props: ', props);
  return (
    <div>
      <h1>Expense List</h1>
      {
        (props.expenses.length === 0) ? (
          <p>There are no expenses.</p>
        ) : (
          props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense} />;
          })
        )
      }
    </div>
  );
};

const mapStateToProps = (state) => {
  //console.log('ExpenseList.js mapStateToProps state: ', state);
  return {
    expenses: selectExpenses(state.expenses, state.filters)
    // expenses: state.expenses
    
  };
};

// this export default is for the app and it uses store data
export default connect(mapStateToProps)(ExpenseList);
