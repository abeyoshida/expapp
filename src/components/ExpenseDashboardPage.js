import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import selectExpensesTotal from '../selectors/expenses-total';
import ExpensesSummaryTotal from './ExpensesSummaryTotal';


const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummaryTotal />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
