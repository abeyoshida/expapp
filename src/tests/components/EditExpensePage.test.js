import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expensesDummy from '../fixtures/expenses';

let wrapper, editExpense, removeExpense, history, expense;

beforeEach( () => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };
    expense = expensesDummy[0];
    wrapper = shallow(
        <EditExpensePage 
            editExpense={editExpense} 
            removeExpense={removeExpense}
            history={history} 
            expense={expense}
        />);
});

test('render EditExpensePage correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('handle editExpense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expense);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expense.id, expense);
});

test('handle removeExpense', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({id: expense.id});
});