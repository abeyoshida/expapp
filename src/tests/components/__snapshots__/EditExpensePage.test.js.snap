// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render EditExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 225,
        "createdAt": 0,
        "description": "coffee",
        "id": "1",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
